src/factories.py

```diff
@@ -0,0 +1,70 @@
+import random
+
+import factory
+from django.utils.timezone import now
+from factory import post_generation
+from factory.django import DjangoModelFactory
+
+from competitions.models import Competition, Phase, Submission
+from datasets.models import Data
+from profiles.models import User
+
+
+class UserFactory(DjangoModelFactory):
+    class Meta:
+        model = User
+        django_get_or_create = ('username',)
+
+    username = factory.Faker('user_name')
+    email = factory.LazyAttribute(lambda o: f"{o.username}@example.com")
+    password = factory.PostGenerationMethodCall('set_password', 'admin')
+
+    @post_generation
+    def super_user(self, created, extracted, **kwargs):
+        if extracted:
```

Comment:
what does `extracted` do here? neat

Reply:
extracted is whatever you supply it in the instantiation (if that's what its called) of a factory:
`UserFactory(super_user=True)`
`True` is the value of extracted
Can also do stuff like:
`UserFactory(super_user__is_staff=False)` or something like that and then `is_staff=False` would be present in `kwargs`.

Reply:
Could change `extracted` to `value` or something to make it clearer?